SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialTraceEventDataBuilder.cpp

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/Connection/SpatialTraceEventDataBuilder.h"
+
+#include <inttypes.h>
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Utils/SpatialActorUtils.h"
+
+#include "Math/UnrealMathUtility.h"
+
+DEFINE_LOG_CATEGORY(FSpatialTraceEventDataBuilderLog);
+
+namespace SpatialGDK
+{
+// ---- FStringCache ----
+
+int32 FSpatialTraceEventDataBuilder::FStringCache::CombineStrings(const char* A, const char* B)
+{
+	int32 InsertIndex = NextIndex;
+	AddString(A);
+	NextIndex--;
+	AddString(B);
+	return InsertIndex;
+}
+
+int32 FSpatialTraceEventDataBuilder::FStringCache::AddString(const char* String)
+{
+	if (NextIndex >= BufferSize)
+	{
+		return BufferSize - 1;
+	}
+
+	int32 InsertIndex = NextIndex;
+	int32 InitialRemainingSize = BufferSize - NextIndex;
+	FCStringAnsi::Strncpy(&Buffer[NextIndex], String, InitialRemainingSize);
+
+	int32 CharSize = sizeof(char);
+	int32 StringLength = FCStringAnsi::Strlen(String) + CharSize;
```

Comment:
Nit: These units are technically different (number of characters vs number of bytes) - since we assume a character is a single byte from strlen we probably don't need to use sizeof(char)

Reply:
hmm so maybe is should be:

FCStringAnsi::Strlen(&Buffer[NextIndex]) * sizeof(char) + sizeof(char)

Reply:
yeah potentially, I think just assuming 1-byte chars is also totally fine too (we use StringAnsi which is char* so always 1-byte anyway). 1-byte chars are defined by the c/cpp standard